tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/CIEnvironmentVariableTests.cs

```diff
@@ -42,6 +43,22 @@ public static IEnumerable<object[]> GetJsonItems()
             }
         }
 
+        [InlineData("git@github.com:DataDog/dd-trace-dotnet.git", true)]
+        [InlineData("git@git@hub.com:DataDog/dd-trace-dotnet.git", false)]
+        [InlineData("git@git/hub.com:DataDog/dd-trace-dotnet.git", false)]
+        [InlineData("git@git$hub.com:DataDog/dd-trace-dotnet.git", false)]
+        [InlineData("github.com:DataDog/dd-trace-dotnet.git", false)]
+        [InlineData("https://github.com/DataDog/dd-trace-dotnet.git", true)]
+        [InlineData("https://github.com/DataDog/dd-trace-dotnet.dit", false)]
+        [InlineData("https://github.com/DataDog/dd-trace-dotnet.git123", false)]
+        [InlineData("git@gitlab.com:gitlab-org/gitlab.git", true)]
+        [InlineData("https://gitlab.com/gitlab-org/gitlab.git", true)]
+        [SkippableTheory]
+        public void RepositoryPattern(string value, bool expected)
+        {
+            Assert.Equal(expected, Regex.Match(value, CIEnvironmentValues.RepositoryUrlPattern).Length != value.Length);
```

Comment:
Can't we simplify this? 🤔 
```suggestion
            Regex.IsMatch(value, CIEnvironmentValues.RepositoryUrlPattern)).Should().Be(expected);
```

Reply:
`IsMatch` returns `true` with partial match of groups in the pattern. That's why the validation compare the length of the match with the value length.

Reply:
👍  Might be worth adding a comment to that effect then 😉